### DISCUSSION TITLE
AttributedString attributes aren't parsed to STTextView attributes

### DISCUSSION THREAD

**Discussion by BugorBN** [krzyzanowskim/STTextView#72] [Questions]
Swift 6.0
Xcode 16.0

Using more complex AttributedString with SwiftUI doesn't work at all.

Instead of ` text: AttributedString = ""` it'll be nice to have more complex constructions work

```
let attributedString = AttributedString()
attributedString.foregroundColor = .red
attributedString.backgroundColor = .yellow

text = attributedString
```

Otherwise I don't know why you decided to use AttributedString without supporting the rest of attributes

P.S.

I saw an interface for UIKIt and it's more than enough. There is no equivalent for the SwiftUI interface

**Comment by krzyzanowskim** [comment]
```swift
attributedString.foregroundColor = .red
attributedString.backgroundColor = .yellow
```

unfortunately these are from SwiftUI attributes scope, that is different than foundation, UIKit and AppKit scope. Identifiers are String based.

that is, it create different attribute indentifiers, because SwiftUI doesn't like anyone
```
"SwiftUI.BackgroundColor" = yellow;
"SwiftUI.ForegroundColor" = red;
```

the unfortunate approach is 

```swift
var attributedString = AttributedString("lalamido")
attributedString.uiKit.foregroundColor = .red
attributedString.uiKit.backgroundColor = .yellow
```

I wish there was a converter for the common attributes. I just hate it.

**Reply by BugorBN** [reply]
Thanks, this was usefull

**Reply by krzyzanowskim** [reply]
I should implement that conversion from AttributedString to NSAttributedString eventually to make it work. one day.
